### PR TITLE
Updating if an origin other than human is used will cancel the command!

### PR DESCRIPTION
**Pull request by gengyoubo**
Note that I changed the forge to version 40.2.4 (No conditional variables have been applied to tf yet)

### CODE REVIEW

**Comment by gengyoubo** [comment]
By the way where is the java file for latex transfur?

**Comment by cecnull1** [comment]
而且，也有自然的兽化情况啊……

**Comment by cecnull1** [comment]
我的建议是：搞一个配置文件
因为有些人就喜欢**Origin+Transfur双重益处**，但也有些人会认为这**破坏游戏平衡性**，或者是**破坏模组间兼容性**

My suggestion is to have a configuration file.
Because some people like the **dual benefits of Origin + Transfur**, but others may think it **breaks game balance** or **causes mod compatibility issues**.

**Comment by gengyoubo** [comment]
> 而且，也有自然的兽化情况啊……
> 
> 

这个我找不到......

**Comment by gengyoubo** [comment]
> 我的建议是：搞一个配置文件 因为有些人就喜欢**Origin+Transfur双重益处**，但也有些人会认为这**破坏游戏平衡性**，或者是**破坏模组间兼容性**
> 
> 我的建议是有一个配置文件。因为有些人喜欢 **Origin + Transfur 的双重好处**，但其他人可能认为它**破坏了游戏平衡**或**导致 mod 兼容性问题**。

添加了！

**Comment by gengyoubo** [comment]
“Dr. K., our experiments using other than human beings have revealed an interesting phenomenon.”
“Tell me about it.”
“Although it's only a small portion, it actually has the ability before latex and transfur, and what they have in common is that they all retained their consciousness, but ......”
“But what?”
“Some creatures have side effects if they TRANSFURRED to use their abilities, such as ...... (blurring)”
“Interesting, let's organize all these records into a (Indistinct)”
“Okay.”
Mutated Latex Experiment - 1

**Review by LtxProgrammer — CHANGES_REQUESTED** [review]
A lot of issues in the PR, and I would generally prefer if comments are in English. I've mentioned to you before how this mod uses `ChangedCompatibility.isPlayerUsedByOtherMod()` to determine if the player should or shouldn't be able to transfur given another mod's conditions. It would be wise to write the origin compatibility to fit within the `isPlayerUsedByOtherMod()` utility function.

Lastly, a small issue I have with the structure of this code is the name of the package `net.ltxprogrammer.changed.VariantCheck.*`, as the capitalization doesn't fit alongside neighboring packages

File: src/main/java/net/ltxprogrammer/changed/VariantCheck/VariantConfig.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import net.minecraftforge.fml.config.ModConfig;
+
+public class VariantConfig {
+    public final ForgeConfigSpec.ConfigValue<Boolean> openOrigin;
+
+    // Constructor that accepts ModLoadingContext
+    public VariantConfig(ModLoadingContext context) {
+        ForgeConfigSpec.Builder builder = new ForgeConfigSpec.Builder();
+        builder.comment("Enabling this config will allow you to have both origin and latex variants. (default is false)");
+        openOrigin = builder.define("openOrigin", false);
```
[the remark this answers is not in the document]
Comment (gengyoubo) [reply src/main/java/net/ltxprogrammer/changed/VariantCheck/VariantConfig.java]: May add #1045 as a follow-up

File: src/main/java/net/ltxprogrammer/changed/VariantCheck/PatienceCompatibility.java
[the remark this answers is not in the document]
Comment (gengyoubo) [reply src/main/java/net/ltxprogrammer/changed/VariantCheck/PatienceCompatibility.java:1]: You're right, so where's the code for transfur?

**Pull request closed without merging**